pkg/k8s/network_policy_v1beta.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+// FIXME Remove this file in k8s 1.8
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/policy/api"
+
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+// ExtractPolicyNameDeprecated extracts the name of policy name
+func ExtractPolicyNameDeprecated(np *v1beta1.NetworkPolicy) string {
+	policyName := np.Annotations[AnnotationName]
+	if policyName == "" {
+		policyName = np.Name
+	}
+
+	return fmt.Sprintf("%s=%s", PolicyLabelName, policyName)
+}
+
+// ParseNetworkPolicyDeprecated parses a k8s NetworkPolicy and returns a list of
+// Cilium policy rules that can be added
+func ParseNetworkPolicyDeprecated(np *v1beta1.NetworkPolicy) (api.Rules, error) {
+	ingress := api.IngressRule{}
+	namespace := ExtractNamespace(&np.ObjectMeta)
+	for _, iRule := range np.Spec.Ingress {
+		// Based on NetworkPolicyIngressRule docs:
+		//   From []NetworkPolicyPeer
+		//   If this field is empty or missing, this rule matches all
+		//   sources (traffic not restricted by source).
+		if iRule.From == nil || len(iRule.From) == 0 {
+			all := api.NewESFromLabels(
+				labels.NewLabel(labels.IDNameAll, "", labels.LabelSourceReserved),
+			)
+			ingress.FromEndpoints = append(ingress.FromEndpoints, all)
+		} else {
+			for _, rule := range iRule.From {
+				// Only one or the other can be set, not both
```

Comment:
This is saying that only rule.PodSelector or rule.NamespaceSelector can be set, but not both; should we check to see if both are set, and error out accordingly?